### PR TITLE
Adds documentation for running a distributed CPU training job using a Kubernetes cluster

### PR DESCRIPTION
**Pull request by dmsuehir**
This is a placeholder PR with a draft of documentation.

### CODE REVIEW

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+[Kubeflow](https://www.kubeflow.org) simplifies the process of deploying machine learning workloads to Kubernetes
+clusters. For this example, the [Kubeflow PyTorchJob training operator](https://www.kubeflow.org/docs/components/training/pytorch/)
+is used to deploy the distributed training job to the cluster. For efficient training on CPUs,
+[Intel® Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch) and
+[Intel® oneCCL Bindings for PyTorch](https://github.com/intel/torch-ccl) libraries are used.
+
+## Cluster setup
+
+This guide assume that you already have a Kubernetes clusters with multiple CPUs that will be used to run the
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: `clusters` -> `cluster`

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+distributed training job. Use `kubectl get nodes` to see a list the nodes that are available on your cluster. Before
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: `to see a list the nodes` -> `to see a list of the nodes`

Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: I'd also add a note here that:
```
I'd add a note here and before asking user to run `kubectl get nodes` like this:
```
If Role-based access control (RBAC) enabled on the cluster, listing nodes and many other cluster wide commands, require specific roles to be granted to the user.```
```

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+running the distributed training job on the cluster, Kubeflow needs to be installed and there needs to be a storage
+location that can be used for the dataset and model files.
+
+### Kubeflow Install
+
+Follow the [Kubeflow installation](https://www.kubeflow.org/docs/started/installing-kubeflow/) guide to deploy the
+Kubeflow resources to your cluster. To verify that the PyTorch custom resource has been deployed to your cluster, use
+`kubectl get crd` and ensure that the output includes `pytorchjob.kubeflow.org` like the sample output below:
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: Or maybe reword it like:
use `kubectl get crd pytorchjobs.kubeflow.org` and verify the output is similar to:

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+typically need to include PyTorch, transformers, Intel Extension for PyTorch, Intel oneCCL Bindings for PyTorch, and
+OpenSSH to communicate between the containers. This container can be used as a base and then you can include your
+training script on top of that.
+
+<!--
+TODO: Maybe we can contribute a dockerfile here that is updated and includes everything needed to run distributed jobs.
+      There's a docker directory with dockerfiles for CPU and GPU, however the CPU dockerfiles are outdated and don't
+      include things like IPEX. Also, Hugging Face has transformers images published to DockerHub, but again the ones
+      for CPU are outdated (haven't been updated in 2 years) -- maybe we can find out why.
+-->
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md:89]: Sounds good and I can work on those Dockerfiles.

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+ * The number of replicas (workers)
+ * The types of resources (node label, memory, and CPU) needed for each worker
+ * The image/tag for the Docker container to use
+ * Environment variables
+ * The python script and it's parameters that will be used to run the training job
+ * A volume mount for the PVC
+
+The volume mount defines a path where the PVC will be mounted in the container for each worker pod. This location can be
+used for the dataset, checkpoint files, and the saved model after training completes.
+<!-- TODO: What would it look like if a cloud storage bucket was being used instead? -->
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md:148]: I can fill this.
I need to get a s3 bucket going and do a PoC.

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+## Summary
+
+This guide walked through the setup and best known methods for using the
+[Kubeflow PyTorch training operator](https://www.kubeflow.org/docs/components/training/pytorch/)
+to run a distributed training job on a Kubernetes cluster with the
+[question-answering example](https://github.com/huggingface/transformers/tree/main/examples/pytorch/question-answering).
+The PyTorchJob yaml used in this example can be used as a template to run your own script and workload. Swap in your
+container image along with the python script and training parameters to the yaml file.
+
+For a more detailed example, out this [blog post (TBD)](TBD) that uses a Kubernetes cluster to fine tune
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: `out this` -> `checkout this`

**Review by ashahba — CHANGES_REQUESTED** [review]
Thanks Dina!
This is a good start.
My comments are mostly minor.

File: docs/source/en/perf_train_cpu_many_k8s.md [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+
+## Cluster setup
+
+> If [role-based access control (RBAC)](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) is enabled on the
+> cluster, listing nodes and many other cluster wide commands require specific roles to be granted to the user.
+
+<!--
+TODO: In the case where RBAC prevents a user from seeing the nodes, what alternative command can we provide to allow the
+      user to verify that their kubectl is properly configured. kubectl get pods? kubectl get ns?
+-->
```
Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md:36]: In that case, things could get complicated and basically available commands to the user are unknown.
I think the best way to get around this is to mention that:
Please consult cluster admin to get a list of nodes available to the user's group.

Comment (ashahba) [inline docs/source/en/perf_train_cpu_many_k8s.md]: In that case, things could get complicated and basically available commands to the user are unknown.
I think the best way to get around this is to mention that:
Please consult cluster admin to get a list of nodes available to the user's group.

**Comment by dmsuehir** [comment]
@sywangyi This is the PR mentioned in the meeting with documentation for running distributed training on Kubernetes

**Comment by dmsuehir** [comment]
Closing, since we switched this to e part of the distributed CPU doc instead of a separate doc.
